src/Controller/CRUDController.php

```diff
@@ -5,18 +5,87 @@
+use Zicht\Bundle\AdminBundle\Event\AdminEvents;
+use Zicht\Bundle\AdminBundle\Event\ObjectDuplicateEvent;
+use Symfony\Component\PropertyInfo\Extractor\ReflectionExtractor;
+use Symfony\Component\PropertyInfo\PropertyInfoExtractor;
 
 /**
  * Provides some basic utility functionality for admin controllers, to be supplied as an construction parameter
  */
 class CRUDController extends BaseCRUDController
 {
+    /** @var string[] */
+    protected $overrideExcludedProperties = ['copiedFrom'];
+
+    /**
+     * Override content of an original page with content of a new page and remove new page
+     *
+     * @return RedirectResponse
+     */
+    public function overrideAction()
+    {
+        $id = $this->getRequest()->get($this->admin->getIdParameter());
+        $object = $this->admin->getObject($id);
+        $originalObject = $object->getCopiedFrom();
+
+        // Override properties without association mapping/relations (like title, introduction etc)
+        $propertyAccessor = PropertyAccess::createPropertyAccessor();
+        $classMetadata = $this->getDoctrine()->getManager()->getClassMetadata(get_class($object));
+
+        $fieldMappings = $classMetadata->getFieldNames();
+
+        foreach ($fieldMappings as $property) {
+            if (in_array($property, $this->overrideExcludedProperties)) {
+                continue;
+            }
+            $value = $propertyAccessor->getValue($object, $property);
+
+            $propertyAccessor->setValue($originalObject, $property, $value);
+        }
+
+        // Override properties with association mapping/relations: meaning the property is also an object and there's an oneToOne, oneToMany or manyToMany relation
+        // between the object and the property. To override properties in case of an oneToMany relation (e.g. one page can have many contentitems) the contentitems
+        // have to be cloned and set to the "receiving" page before the "providing" page gets deleted, because of the cascade remove option.
+        $associationMappings = $classMetadata->getAssociationNames();
+
+        foreach ($associationMappings as $property) {
+            if (in_array($property, $this->overrideExcludedProperties)) {
+                continue;
+            }
+            $values = $propertyAccessor->getValue($object, $property);
+            $mappingType = $classMetadata->getAssociationMappings()[$property]['type'];
+
+            if ($mappingType === ClassMetadata::ONE_TO_MANY) {
+                $clonedValues = [];
+                foreach ($values as $value) {
+                    $clonedValues[] = clone $value;
+                }
+                $propertyAccessor->setValue($originalObject, $property, $clonedValues);
+            } else {
+                $propertyAccessor->setValue($originalObject, $property, $values);
+            }
+        }
+
+        $objectManager = $this->getDoctrine()->getManager();
+        $objectManager->persist($object);
```

Comment:
Dit moet zijn:
```suggestion
        $objectManager->persist($originalObject);
```

Ik vroeg of je `$originalObject` expliciet kon persisten, ondanks dat de Entity Manager deze in de gaten zou houden en bij een flush toch wel zou persisten, omdat ik het gevoel had dat het beter was om het expliciet te doen. Nou het blijkt dus dat dat automatisch flushen dus niet altijd het geval is: https://www.doctrine-project.org/projects/doctrine-orm/en/2.7/reference/change-tracking-policies.html#deferred-explicit

Ik zie nog wel eens entities met tracking policy "deferred explicit": https://github.com/zicht/musisenstadstheater.nl/blob/release/1.1.x/src/Zicht/Bundle/MusisSiteBundle/Entity/Page/ContentPage.php#L18 bijv.

Reply:
Oke check, ik had het verkeerd begrepen/gelezen :)